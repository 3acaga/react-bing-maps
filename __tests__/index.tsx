import * as React from "react";
import { shallow, render, mount } from "enzyme";

import ReactBingMaps, { Pushpin, Polyline, Layer, Infobox } from "../src";

describe("react-bing-maps", () => {
  it("components are defined", () => {
    expect(ReactBingMaps).toBeDefined();
    expect(Pushpin).toBeDefined();
    expect(Polyline).toBeDefined();
    expect(Layer).toBeDefined();
    expect(Infobox).toBeDefined();
  });

  it("should initialize the map", async () => {
    let map: any;

    let promise = new Promise((resolve) => {
      map = mount(
        <ReactBingMaps
          apiKey=""
          onMapInit={() => {
            resolve();
          }}
        />
      );
    });

    await promise;

    expect(map.exists("div#react-bing-maps")).toBe(true);
    expect(map.exists("canvas#tileCanvasd")).toBe(true);
    expect(map.exists("canvas#labelCanvasId")).toBe(true);
    expect(map.exists("#not-react-bing-maps")).toBe(false);
  });
});
