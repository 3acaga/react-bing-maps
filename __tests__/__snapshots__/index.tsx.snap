// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`react-bing-maps should initialize the map 1`] = `ShallowWrapper {}`;

exports[`should initialize the map 1`] = `ShallowWrapper {}`;
